stixcore/products/product.py

```diff
@@ -704,7 +716,43 @@ def from_level0(cls, l0product, parent=''):
                  ssid=l0product.ssid,
                  control=l0product.control,
                  data=l0product.data,
-                 idb_versions=l0product.idb_versions)
+                 idb_versions=l0product.idb_versions,
+                 comment=l0product.comment,
+                 history=l0product.history)
+
+        if l1.service_type == 21 and l1.service_subtype == 6 and l1.ssid in (20, 21, 22, 23, 24):
+            # Time and count arrays were off by one so need to shift before FSW version
+            # https://github.com/i4Ds/STIXCore/issues/286
+            idbs = sorted([tuple(map(int, k.split('.'))) for k in l0product.idb_versions.keys()])
+            if len(idbs) > 1:
+                raise ValueError('More than one IDB found in BSD product')
+
+            # Problem only occurred before 2.26.36 version of FSW problem was fixed in
+            if idbs[0] < (2, 26, 36) and len(l1.data) > 1:
+                # Check if request was at min configured time resolution
+                if (l1.utc_timerange.start.datetime < MIN_INT_TIME_CHANGE
+                    and l1.data['timedel'].as_float().min() == 1*u.s) or \
+                        (l1.utc_timerange.start.datetime >= MIN_INT_TIME_CHANGE
+                         and l1.data['timedel'].as_float().min() == 0.5*u.s):
+                    correct_counts = l1.data['counts'][1:]
+                    correct_count_errs = l1.data['counts_err'][1:]
+                    correct_triggers = l1.data['triggers'][1:]
+                    correct_trigger_errs = l1.data['triggers_err'][1:]
+                    correct_times = l1.data['time'][1:]
+                    correct_durations = l1.data['timedel'][1:]
+                    correct_rcr = l1.data['rcr'][1:]
+                    l1.data = l1.data[0:-1]
+                    l1.data['counts'] = correct_counts
+                    l1.data['counts_err'] = correct_count_errs
+                    l1.data['triggers'] = correct_triggers
+                    l1.data['triggers_err'] = correct_trigger_errs
+                    l1.data['time'] = correct_times
+                    l1.data['timedel'] = correct_durations
+                    l1.data['rcr'] = correct_rcr
+                    l1.history.append('Time and count arrays were shifted to fix offset')
+                else:
+                    l1.comment.append('Time and count arrays offset not fixed'
+                                      ' as possibly summed on board')
```

Comment:
i know about that but somehow i think other users might not could we add a link to some kind of online documentation / git issue/ dpdd that will provide more background
?

Reply:
Yea I think we need a read me with the data release summarising the data and know issues.